### PR TITLE
[Bug] Fix DoReMi's DRO Loss

### PR DESCRIPTION
**Pull request by xrsrke**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -162,3 +162,4 @@ cython_debug/
 .vscode
 
 checkpoints/
+wandb/*
```
Comment (3outeille) [inline .gitignore:165]: shouldn't be push to main

File: src/nanotron/trainer.py [hunk 2]
```diff
@@ -264,13 +261,35 @@ def post_training(self):
...
+        if not hasattr(self.config, "data_stages") or self.config.data_stages is None:
+            if self.current_dataloader is None:
+                if isinstance(dataloaders, tuple):
+                    dataloader = dataloaders[0]
+                else:
+                    dataloader = dataloaders
+                self.current_dataloader = sanity_check_dataloader(
+                    dataloader=dataloader, parallel_context=self.parallel_context, config=self.config
+                )
+            return
```
Comment (3outeille) [inline src/nanotron/trainer.py:286]: return is not necessary here

File: src/nanotron/trainer.py [hunk 3]
```diff
@@ -264,13 +261,35 @@ def post_training(self):
...
+        elif isinstance(dataloaders, Generator):
+            # TODO(xrsrke): this is a hacky way to handle DoReMi's dataloader
+            # remove this in the next PR
+            self.current_dataloader = dataloaders
+            return
```
Comment (3outeille) [inline src/nanotron/trainer.py:291]: same as above

File: src/nanotron/trainer.py [hunk 4]
```diff
@@ -264,13 +261,35 @@ def post_training(self):
...
+                else:
+                    dataloader = dataloaders
+                self.current_dataloader = sanity_check_dataloader(
+                    dataloader=dataloader, parallel_context=self.parallel_context, config=self.config
+                )
+            return
+        elif isinstance(dataloaders, Generator):
+            # TODO(xrsrke): this is a hacky way to handle DoReMi's dataloader
+            # remove this in the next PR
+            self.current_dataloader = dataloaders
```
Comment (3outeille) [inline src/nanotron/trainer.py:290]: should we do sanity_check_dataloader here as well ?

File: .gitignore (hunk 1 above)
Comment (xrsrke) [reply .gitignore:165]: this is gitignore though?

File: src/nanotron/trainer.py (hunk 3 above)
Comment (xrsrke) [reply src/nanotron/trainer.py:291]: nah, i stop the function execution

File: src/nanotron/trainer.py (hunk 4 above)
Comment (xrsrke) [reply src/nanotron/trainer.py:290]: nah, it's already a generator

**Pull request merged**